File: examples/demo_hdf5_timefreq.m
%% Test data file
% The following file contains data acquired from the spectrometer.
testfile = './data/acq_data_100_measurements.hdf5';

%% Data file format
% Data is stored in the HDF5 file in a simple format.  
% 
% Each Acquisiton is stored in a group, allowing multiple acquisitons
% to be stored in one file.  Each acquisiton group contains three datasets:
% 
% * *fftdata:* Measurement results
% * *msrmnt_num:* Measurement number (id)
% * *num_averages:* Number of accumulated spectra.
%
% Each acquisiton group also contains a 'count' attribute indicating the
% number of measurements recorded for the acquisition.
%
% The contents of the testfile serves as an example:
h5disp(testfile)

%% Reading data from file

% Get name of first acquisition group
info = h5info(testfile);
groupName = info.Groups(1).Name;
disp(['Found acquisiton ' groupName]);

% Get acquired data
measurement_count = h5readatt(testfile,groupName,'count');
fftdata      = h5read(testfile,[groupName '/fftdata']);
msrmt_num    = h5read(testfile,[groupName '/msrmt_num']);
num_averages = h5read(testfile,[groupName '/num_averages']);

%% Plot results
[NfftDiv2,~]  = size(fftdata);
F = (0:(NfftDiv2-1)) * 1000./NfftDiv2;
T = 0:measurement_count-1;
waterfall(T, F, 10*log10(fftdata))
xlabel('Time')
ylabel('Frequency (MHz)')
axis tight
view(90, 90)

File: examples/demo_hdf5_timefreq.py
#------------------------------------------------------------------------------
# Copyright (c) 2016, DSPlogic, Inc.  All Rights Reserved.  
# 
# RESTRICTED RIGHTS
# Use of this software is permitted only with a software license agreement.
#
# Details of the software license agreement are in the file LICENSE.txt, 
# distributed with this software.
#------------------------------------------------------------------------------
from __future__ import (division, print_function, absolute_import)

""" Demonstrate time/frequency plot using HDF5 acquisition format

This program demonstrates:

    - Reading recorded instrument data from a HDF5 log file
    - Surface contour of time vs frequency

"""

import os, logging
from pyspectro.applib.datalogger import SpectrumDataReader
from plot_helpers import plot_raw_data, plot_waterfall

#: Configure logging
logger = logging.getLogger(__name__)
logging.basicConfig(level=logging.DEBUG,
                    format="%(relativeCreated)5d %(threadName)10s %(levelname)-8s %(message)-60s  <%(name)-15s>")

""" Ask user to select HDF5 file """
filename = './data/acq_data_100_measurements.hdf5' 

#: Create HDF5 reader object
rdr = SpectrumDataReader(filename)

#: Number of acquisitons in file
logger.info('Found %s acquisitions' % len(rdr.acquisitions))

#: Inspect first acquisition            
acq1 = rdr.acquisitions[0]
logger.info('Acquisiton ID: %s' % acq1['name'])         
logger.info('Number of measurements: %s' % acq1['msrmnt_count'])

#: Plot results of *first* measurement in acquisiton
rawfftdata  = acq1['fftdata']         
numAverages = acq1['num_averages']
msrmnt_idx  = acq1['msrmnt_idx'] #: This value can be used to detect "dropped" measurement recordings

logger.info('Plotting measurement %s' % msrmnt_idx[0])
plot_raw_data(rawfftdata[0], numAverages[0])
plot_waterfall(rawfftdata , numAverages, msrmnt_idx )

